### PR TITLE
Fix LiteLLM streaming usage and reasoning telemetry

### LINKED ISSUE #324: LiteLLM streaming instrumentation drops usage and reasoning chunks
## Problem

The LiteLLM completion streaming wrappers incompletely reconstruct streamed responses:

1. They do not collect `usage` from stream chunks. OpenAI-compatible providers commonly return usage in a final chunk with `choices: []` when `stream_options={"include_usage": true}` is enabled. Since that chunk has no choices, the current accumulator ignores it and the finished Laminar span has no token attributes.
2. They collect `delta.content` and `delta.tool_calls`, but not streamed reasoning fields such as `delta.reasoning_content`. As a result, reasoning can be present in the provider stream but absent from Laminar's reconstructed output and downstream trajectory exports.

This affects both sync and async completion streaming in:

`src/lmnr/opentelemetry_lib/opentelemetry/instrumentation/litellm/wrappers/completions/streaming.py`

### Usage

In a production Kimi K3 sample, successful streaming calls had assistant content, tool calls, response IDs, and successful status, but their Laminar spans reported zero input/output tokens. The behavior was route-dependent: 18,796 direct `kimi-k3` streaming calls had no usage while 503 calls through a route that produced a complete response had usage.

### Reasoning

Kimi K3 calls are configured with high reasoning effort, but only about 5% of retained agent steps in a large exported sample had non-empty `reasoning_content`. Raw non-streaming response objects demonstrate that Kimi/LiteLLM can expose reasoning as top-level `reasoning_content` and provider-specific `reasoning`/`reasoning_details`. The streaming accumulator currently has no code to preserve reasoning deltas, making dropped streamed reasoning a likely contributor to the low observed rate.

## Minimal usage reproduction

Pass these chunks through `process_completion_streaming_response` and inspect the finished span:

```python
chunks = [
    {
        "id": "chatcmpl-test",
        "model": "kimi-k3",
        "choices": [{"index": 0, "delta": {"content": "Hello"}}],
    },
    {
        "id": "chatcmpl-test",
        "model": "kimi-k3",
        "choices": [],
        "usage": {
            "prompt_tokens": 5000,
            "completion_tokens": 100,
            "total_tokens": 5100,
            "prompt_tokens_details": {"cached_tokens": 4000},
        },
    },
]
```

Expected attributes:

- `gen_ai.usage.input_tokens = 5000`
- `gen_ai.usage.output_tokens = 100`
- `llm.usage.total_tokens = 5100`
- `gen_ai.usage.cache_read_input_tokens = 4000`

Actual: none of these attributes are present.

## Minimal reasoning reproduction

```python
chunks = [
    {
        "id": "chatcmpl-test",
        "model": "kimi-k3",
        "choices": [{
            "index": 0,
            "delta": {
                "role": "assistant",
                "reasoning_content": "I should inspect ",
            },
        }],
    },
    {
        "id": "chatcmpl-test",
        "model": "kimi-k3",
        "choices": [{
            "index": 0,
            "delta": {
                "reasoning_content": "the repository.",
                "content": "I'll inspect it.",
            },
            "finish_reason": "stop",
        }],
    },
]
```

Expected reconstructed assistant message:

```json
{
  "reasoning_content": "I should inspect the repository.",
  "content": "I'll inspect it."
}
```

Actual: content is retained, but reasoning is absent.

## Suggested fix

- Accumulate the last non-null `usage` object from streaming chunks and apply the same input/output/total/cache token mapping used for non-streaming completions when finalizing the span.
- Accumulate `delta.reasoning_content`; consider normalized fallbacks such as `delta.reasoning` and `delta.thinking` where LiteLLM/provider adapters expose them.
- Include usage and reasoning in the reconstructed raw rollout response.
- Add sync and async regression tests for both a final empty-choice usage chunk and multi-chunk reasoning deltas.

PR #325 currently addresses the usage portion and should be extended to cover reasoning.

_This issue was drafted and updated by an AI agent on behalf of Graham Neubig._

### PR DESCRIPTION
**Pull request by neubig**
## Summary

- retain the last non-null `usage` object while consuming LiteLLM completion streams
- write input, output, total, cache-read, and cache-creation token attributes when reported
- accumulate streamed reasoning deltas (`reasoning_content`, with normalized `reasoning`/`thinking` fallbacks) into reconstructed output messages
- include usage and reasoning in reconstructed raw rollout responses
- avoid reporting unknown cache usage as confirmed zero
- add synchronous and asynchronous regression tests for usage-only chunks and multi-chunk reasoning

Fixes #324.

## Testing

```text
uv run pytest tests/test_litellm.py \
  -k 'streaming_records_usage_only_chunk or streaming_preserves_reasoning_deltas or streaming_omits_unknown_cache_usage' \
  -q

5 passed, 26 deselected
```

```text
uv tool run ruff check \
  src/lmnr/opentelemetry_lib/opentelemetry/instrumentation/litellm/wrappers/completions/streaming.py

All checks passed!
```

_This pull request was created and updated by an AI agent on behalf of Graham Neubig._

---

> [!NOTE]
> **Low Risk**
> Instrumentation-only change to LiteLLM stream span attributes; no auth, data-store, or request-path behavior changes.
> 
> **Overview**
> LiteLLM streaming spans now keep token usage and reasoning text that previously dropped off the reconstructed completion.
> 
> The stream accumulator retains the last non-null `usage` object (including usage-only final chunks) and writes input, output, total, cache-read, and cache-creation attributes when those fields are present. Missing cache details are left unset instead of recorded as zero.
> 
> Streamed reasoning deltas (`reasoning_content`, with `reasoning`/`thinking` fallbacks) are concatenated into output messages and raw rollout responses. Sync and async tests cover usage-only chunks, reasoning accumulation, and omitted cache attributes. Bumps the SDK to `0.7.60`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 781134a1cd4c76751fc4f5625d52fdbee7bf9124. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/lmnr/opentelemetry_lib/opentelemetry/instrumentation/litellm/wrappers/completions/streaming.py [hunk 1]
```diff
@@ -84,6 +91,33 @@ def _set_accumulated_attributes(
...
+                usage.get("cache_creation_input_tokens", 0),
+            )
+            set_span_attribute(
+                span, "gen_ai.usage.cache_read_input_tokens", cache_read_tokens
+            )
+            set_span_attribute(
+                span,
+                "gen_ai.usage.cache_creation_input_tokens",
+                cache_creation_tokens,
+            )
```
[the remark this answers is not in the document]
Comment (neubig) [reply src/lmnr/opentelemetry_lib/opentelemetry/instrumentation/litellm/wrappers/completions/streaming.py]: Fixed in 188659c: cache-read/cache-creation attributes are now emitted only when the provider usage payload actually reports the corresponding field, so missing cache details are no longer recorded as zero. This reply was generated by an AI agent on behalf of Graham Neubig.
